Add Toggletip (#6154)
jpmorganchase/salt-ds: site/docs/components/toggletip/accessibility.mdx

---
title:
  $ref: ./#/title
layout: DetailComponent
sidebar:
  exclude: true
data:
  $ref: ./#/data
---

## Best practices

- Ensure the trigger has a clear, descriptive, accessible name that is associated to what it's related to (e.g. a section header, an image, etc.) so that users understand the context of the information provided in the tooltip.
- Don't user a timer to automatically dismiss the toggletip.
- Ensure the toggletip trigger is visually closely associated with the content it describes so users can easily associate them.

## Keyboard interactions

<KeyboardControls>
<KeyboardControl keyOrCombos="Tab">

- If focus is before the toggletip trigger, Tab moves focus onto the trigger.
- If the trigger has focus, Tab moves focus onto the next interactive elements in the tab order.
- When the toggletip panel is open and focused, Tab moves focus to the next interactive element in the panel.
  - If the panel only contains text or focus is on the last interactive element in the panel, Tab closes the panel and moves focus to the next interactive element in the tab order.

</KeyboardControl>
<KeyboardControl keyOrCombos="Shift + Tab">

> - If focus is on an interactive element immediate after the toggletip trigger, Shift + Tab moves focus onto the trigger.

Author: I've checked, essentially the current docs use this action for combined actions e.g. Shift + Tab, but use the actual key when there's only one

> - If the trigger has focus, Shift + Tab moves focus to the previous interactive element in the tab order.

Reviewer: ```suggestion
- If the trigger has focus, this action moves focus onto the previous interactive element in the tab order.
```

- When the toggletip panel is open and focus in inside, Shift + Tab moves focus to the previous interactive element in the panel.
  - If there is no previous interactive element in the panel, focus moves to the panel.
  - If the panel has focus, this action moves focus back to the trigger but doesn't close the panel.

</KeyboardControl>
<KeyboardControl keyOrCombos="Enter / Space">

- When focus is on the toggletip trigger, Enter or Space opens the panel and moves focus to the panel.

</KeyboardControl>
<KeyboardControl keyOrCombos="Escape">

> - When focus is on or inside the toggletip trigger or panel, Escape closes the panel and keeps/moves focus back to the toggletip trigger.

Reviewer: ```suggestion
- When focus is on or inside the toggletip trigger or panel, this action closes the panel and keeps or moves focus back to the toggletip trigger.
```
